app/src/main/java/com/epam/bet/viewmodel/SubscribeViewModel.kt

```diff
@@ -0,0 +1,104 @@
+package com.epam.bet.viewmodel
+
+import android.app.Application
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Log
+import androidx.lifecycle.AndroidViewModel
+import com.epam.bet.entities.Follower
+import com.epam.bet.extensions.showToast
+import com.google.firebase.firestore.FirebaseFirestore
+import com.firebase.ui.firestore.FirestoreRecyclerOptions
+import com.google.firebase.firestore.Query
+
+class SubscribeViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val db = FirebaseFirestore.getInstance()
+    private val users = db.collection("users")
+    private lateinit var sharedPreferences: SharedPreferences
+
+    fun isItMyEmail(email: String): Boolean {
+        sharedPreferences = getApplication<Application>().applicationContext.getSharedPreferences("BetAppSettings", Context.MODE_PRIVATE)
+        val myEmail = sharedPreferences.getString("email", "none")
+        return email == myEmail
+    }
+
+    fun isIFollow(email: String, context: Context?): Boolean {
+        sharedPreferences = getApplication<Application>().applicationContext.getSharedPreferences("BetAppSettings", Context.MODE_PRIVATE)
+        val userEmail = sharedPreferences.getString("email", "none")
+        val emailList = mutableListOf<String>()
+        var isSuccess = false
+        users.document(userEmail!!).collection("i_follow").get()
+                .addOnSuccessListener { documents ->
+                    for (document in documents) {
+                        emailList.add(document.data.get("email").toString())
+                    }
+                    isSuccess = true
+                }
+
+        if (isSuccess) {
+            if (email in emailList) {
+                context?.showToast("You're already following this user!")
+                return true
+            }
+        } else {
+            context?.showToast("Problem with connection !")
+        }
+        return false
+    }
+
+    fun addIFollow(email: String, successListener: (email: String) -> Unit, failListener: () -> Unit) {
+        val myEmail = sharedPreferences.getString("email", "none")
+        val myName = sharedPreferences.getString("name", "none")
+
```

Comment:
myEmail, myName -> bullshit naming

Reply:
Why? We have concepts like "I follow" in the code. In this context, it is clear that "my email" equals "current user email"